python/ray/experimental/workflow/api.py

```diff
@@ -145,4 +146,26 @@ def get_output(workflow_id: str) -> ray.ObjectRef:
     return execution.get_output(workflow_id)
 
 
-__all__ = ("step", "virtual_actor", "resume", "get_output", "get_actor")
+def list_all(status: Optional[WorkflowStatus] = None
```

Comment:
Should we have a limit by default?

Reply:
```suggestion
def list_all(filter_status: Set[WorkflowStatus] = None
```

Reply:
My concern is that it's called list_all, so if the limit is the default, it might give the user some confusion that it only lists a limited number.

We should add more filters for this one later and maybe support pagination. The API needs better design I think.